integration-tests/soap/src/main/java/org/apache/camel/quarkus/component/soap/it/SoapResource.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.quarkus.component.soap.it.example.GetCustomersByName;
+import org.jboss.logging.Logger;
+
+@Path("/soap")
+@ApplicationScoped
+public class SoapResource {
+
+    private static final Logger LOG = Logger.getLogger(SoapResource.class);
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Path("/marshal")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_XML)
+    public Response marshall(String message) throws Exception {
+        LOG.infof("Sending to soap: %s", message);
+        GetCustomersByName request = new GetCustomersByName();
+        request.setName(message);
+
+        final String response = producerTemplate.requestBody("direct:marshal", request, String.class);
+
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response)
+                .build();
+    }
+
+    @Path("/marshal-soap12")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_XML)
+    public Response marshalSoap12(String message) throws Exception {
+        GetCustomersByName request = new GetCustomersByName();
+        request.setName(message);
+
+        final String response = producerTemplate.requestBody("direct:marshal-soap12", request, String.class);
+
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response)
+                .build();
+    }
+
+    @Path("/unmarshal")
+    @POST
+    @Consumes(MediaType.APPLICATION_XML)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response unmarshal(String message) throws Exception {
+        final GetCustomersByName response = producerTemplate.requestBody("direct:unmarshal", message, GetCustomersByName.class);
+
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response.getName())
+                .build();
+    }
+
+    @Path("/unmarshal-soap12")
+    @POST
+    @Consumes(MediaType.APPLICATION_XML)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response unmarshalSoap12(String message) throws Exception {
+        final GetCustomersByName response = producerTemplate.requestBody("direct:unmarshal-soap12", message,
+                GetCustomersByName.class);
+
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response.getName())
+                .build();
+    }
+
+    @Path("/round")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response round(String message) throws Exception {
+        LOG.infof("Sending to soap: %s", message);
+        GetCustomersByName request = new GetCustomersByName();
+        request.setName(message);
+        final String xml = producerTemplate.requestBody("direct:marshal", request, String.class);
+        LOG.infof("Got response from marshal: %s", xml);
+
```

Comment:
Unless /round is doing something not covered by the above /marshal /unmarshal endpoints, I'd vote for removing it.

Reply:
I was testing if marshaled message could be also un-marshaled. I had some issues with that so I kept it there.